Create documentation for component and related packages (#7)
kyma-incubator/documentation-component: packages/markdown-render-engine/README.md

# Markdown render engine

## Overview

> This render engine is based on [react-markdown](https://github.com/rexxars/react-markdown) component.

Author: `The Markdown render engine is based on the [React Markdown](https://github.com/rexxars/react-markdown) component and allows you to render `.md` files.` ?

## Installation

```bash
npm i @kyma-project/dc-markdown-render-engine
```

> ## Options

Reviewer: ⏫

> **NOTE**: This render engine support all options available in [react-markdown](https://github.com/rexxars/react-markdown#options) component.

> 

Author: Options from `https://github.com/rexxars/react-markdown`? I think link will be better, because package has a lot of configs and options and duplicating content from original Readme is unnecessary.

- Required: `No`
- Type: [`Renderers`](./src/types.ts#L16)
- Default value: `{}`

### `highlightTheme`

Custom styles for highlight of code block. Component use for highlight [`PrismJS`](https://github.com/PrismJS/prism). See example [themes](https://github.com/PrismJS/prism-themes/tree/master/themes).

- Required: `No`
- Type: `any`
- Default value: `{}`

### `headingPrefix`

Prefix for any heading appearing in markdown file.

- Required: `No`
- Type: `string | ((source: Source) => string)`, where [`Source`](./src/interfaces/Source.ts#L14) is type.
- Default value: `""`

### `copyButton`

Custom copy button in code blocks. Component accept only `code` as `string` to copy as prop. 

- Required: `No`
- Type: `ReactNode`
- Default value: `null`

## Plugins

For using plugins available in package, import `plugins` object from package, like:

``` js
import { plugins as markdownPlugins } from "@kyma-project/dc-markdown-render-engine"
```

> **NOTE**: For information how to pass options of plugins, read [this](../../docs/props/plugins.md#passing-globally-options) document.

> 

Author: Ok :)

### `embedVideo`

> Mutation plugin for extract video, inspired by [gatsby-remark-embed-video](https://github.com/borgfriend/gatsby-remark-embed-video). Pattern for embedding video is `{platform}: {url}`, where `{url}` is a url to video, and `{platform}` is one of below:

Author: It's good :)

- `youtube`,
- `vimeo`,
- `videopress`,
- `twitch`,
- `twitchlive`,
- `niconico`,

> **NOTE**: For fully integration with `documentation-component` you also must use [`markdownEmbedVideoParserPlugin`](./src/plugins/embedVideo/parserPlugin.tsx).

- Type: `Mutation`
- SourceTypes: `["markdown", "md"]`

### `frontmatter`

> Mutation and extractor plugin. Mutation for extract file metadata from markdown file. Extractor for remove frontmatter from beginning of file.

Author: But then I must replace extractor name in whole code.

# Markdown render engine

## Overview

> This render engine is based on [react-markdown](https://github.com/rexxars/react-markdown) component.

Reviewer: ```suggestion
The Markdown render engine is based on the [React Markdown](https://github.com/rexxars/react-markdown) component. It allows you to render `.md` files.
```

## Installation

> - by [`npm`](https://www.npmjs.com/)

Reviewer: ```suggestion
- using [`npm`](https://www.npmjs.com/)
```

  ``` bash
  npm i @kyma-project/dc-markdown-render-engine
  ```

> - by [`yarn`](https://yarnpkg.com/en/)

Reviewer: ```suggestion
- using [`yarn`](https://yarnpkg.com/en/)
```

  ``` bash
  yarn add @kyma-project/dc-markdown-render-engine
  ```

## Options

> > **NOTE**: This render engine support all options available in [react-markdown](https://github.com/rexxars/react-markdown#options) component.

Reviewer: ```suggestion
> **NOTE**: The Markdown render engine supports all options available in the [React Markdown](https://github.com/rexxars/react-markdown#options) component.
```

| Name | Required | Type | Default value | Description |
|---|---|---|---|---|

> | `parsers` | No | [`MarkdownParserPlugin[]`](./src/types.ts#L24) | `[]` | Custom parsers for custom content in markdown files. More information about write parsers [here](https://github.com/aknuds1/html-to-react#with-custom-processing-instructions). |

Reviewer: ```suggestion
| `parsers` | No | [`MarkdownParserPlugin[]`](./src/types.ts#L24) | `[]` | Defines custom parsers for custom content in markdown files. Go [here](https://github.com/aknuds1/html-to-react#with-custom-processing-instructions) to read more about write parsers. |
```
Hmm I'm thinking about the write parsers... what they are?:P

> | `customRenderers` | No | [`Renderers`](./src/types.ts) | `{}` | An object where the keys represent the node type and the value is a React component. The object is merged with the default renderers. The props passed to the component varies based on the type of node. See [default](./src/renderers) implementations of renderers. |

Reviewer: We need to change the Description together :)

> | `highlightTheme` | No | `any` | `{}` | Custom styles for highlight of code block. Component use for highlight [`PrismJS`](https://github.com/PrismJS/prism). See example [themes](https://github.com/PrismJS/prism-themes/tree/master/themes). |

Reviewer: ```suggestion
| `highlightTheme` | No | `any` | `{}` | Defines custom styles for highlighting code blocks. |
```
I would remove that part: `Component use for highlight [`PrismJS`](https://github.com/PrismJS/prism). See example [themes](https://github.com/PrismJS/prism-themes/tree/master/themes).`

> | `headingPrefix` | No | `string or ((source: Source) => string)`, where [`Source`](../documentation-component/src/interfaces/Source.ts) is type | `""` | Prefix for any heading appearing in markdown file. |

Reviewer: ```suggestion
| `headingPrefix` | No | `string or ((source: Source) => string)`, where [`Source`](../documentation-component/src/interfaces/Source.ts) is type | `""` | Defines a prefix for any heading appearing in a Markdown file. |
```

Is the type ok?

> | `copyButton` | No | `ReactNode` | `null` | Custom copy button in code blocks. Component accept only `code` as `string` to copy as prop.  |

Reviewer: ```suggestion
| `copyButton` | No | `ReactNode` | `null` | Defines a custom copy button in code blocks.  |
```
We need to rephrase this: `Component accepts only `code` as `string` to copy as prop. `
